### ISSUE TITLE
Build a documentation pipeline

### ISSUE THREAD

**Issue by wekitecs** [WEKIT-ECS/MIRAGE-XR#31]
In GitLab by @BenediktHensen on Jul 17, 2020, 16:17

We should have a publicly available documentation for the project.

**Comment by wekitecs** [comment]
In GitLab by @BenediktHensen on Jul 17, 2020, 16:18

Ideally, we should write documentation in Markdown and generate HTML from this which can be published on a Web page.

**Comment by wekitecs** [comment]
In GitLab by @BenediktHensen on Jul 17, 2020, 16:28

This is how Microsoft is creating documentation for the MRTK:
Their [GitHub Wiki](https://github.com/microsoft/MixedRealityToolkit-Unity/wiki) provides abstract information about the project itself, planning and contribution rules.

They have a folder ["Documentation"](https://github.com/microsoft/MixedRealityToolkit-Unity/tree/mrtk_development/Documentation) in the repository where they keep Markdown files with the documentation. The folder is not included in the shipped package, though.
A CI job generates HTML files using [DocFX](https://dotnet.github.io/docfx/). The files are hosted on a [Web page](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/GettingStartedWithTheMRTK.html).

**Comment by wekitecs** [comment]
In GitLab by @BenediktHensen on Jul 17, 2020, 16:32

DocFX can actually combine Markdown based documentation with in-code XML comments into one Web page, so we get both solutions which are useful for developers and non-developers.

**Comment by wekitecs** [comment]
In GitLab by @BenediktHensen on Jul 17, 2020, 16:49

@klamma @fwild My proposed solution would be to set up a similar pipeline as the one mentioned above. It seems to be a good solution since the documentation files could be shipped with the product, if necessary but they can also be excluded. Moreover, the documentation website consists of HTML-files from Markdown and it can also generate API documentation from the XML-comments in the code. An automated CI pipeline can handle the publishing on a website.

Until we have decided which documentation system we use and until we have set it up, it still makes sense to collect new documentation in this repository's wiki as it automatically provides an in-browser preview. Once the documentation pipeline works, we can copy the wiki's files to the documentation source folder.
